modules/dnsmasq_dhcp/autodetection.go

```diff
@@ -128,3 +217,85 @@ func parseDHCPRangeLine(s string) (r string) {
 
 	return fmt.Sprintf("%s-%s", start, end)
 }
+
+func (dir configDir) findConfigs() ([]string, error) {
+	fis, err := ioutil.ReadDir(dir.path)
+	if err != nil {
+		return nil, err
+	}
+
+	var configs []string
+
+	for _, fi := range fis {
+		if !fi.Mode().IsRegular() {
+			continue
+		}
+
+		// We copy the dnsmasq's logic
+		//
+		// /* ignore emacs backups and dotfiles */
+		// if (len == 0 ||
+		//     ent->d_name[len - 1] == '~' ||
+		//     (ent->d_name[0] == '#' && ent->d_name[len - 1] == '#') ||
+		//     ent->d_name[0] == '.')
+		//    continue;
+
+		name := fi.Name()
+		if strings.HasSuffix(name, "~") ||
+			(strings.HasPrefix(name, "#") && strings.HasSuffix(name, "#")) ||
+			strings.HasPrefix(name, ".") {
+			continue
+		}
```

Comment:
i think this logic deserves its own func `isBadFileName` or something

Reply:
I've added `isValidFileName` and pushed commit with force